app/src/ui/branches/index.tsx

```diff
@@ -57,22 +73,138 @@ export class Branches extends React.Component<IBranchesProps, IBranchesState> {
     this.setState({ selectedBranch })
   }
 
+  private renderTabBar() {
+    if (!this.props.account) {
+      return null
+    }
+
+    if (!enablePreviewFeatures()) {
+      return null
+    }
+
+    let countElement = null
+    if (this.props.pullRequests) {
+      countElement = (
+        <span className="count">{this.props.pullRequests.length}</span>
+      )
+    }
+
+    return (
+      <TabBar
+        onTabClicked={this.onTabClicked}
+        selectedIndex={this.props.selectedTab}
+      >
+        <span>Branches</span>
+        <span className="pull-request-tab">
+          {__DARWIN__ ? 'Pull Requests' : 'Pull requests'}
+
+          {countElement}
+        </span>
+      </TabBar>
+    )
+  }
+
+  private renderSelectedTab() {
+    const tab = this.props.selectedTab
+    switch (tab) {
+      case BranchesTab.Branches:
+        return (
+          <BranchList
+            defaultBranch={this.props.defaultBranch}
+            currentBranch={this.props.currentBranch}
+            allBranches={this.props.allBranches}
+            recentBranches={this.props.recentBranches}
+            onItemClick={this.onItemClick}
+            filterText={this.state.filterText}
+            onFilterKeyDown={this.onFilterKeyDown}
+            onFilterTextChanged={this.onFilterTextChanged}
+            selectedBranch={this.state.selectedBranch}
+            onSelectionChanged={this.onSelectionChanged}
+          />
+        )
+
+      case BranchesTab.PullRequests: {
+        const pullRequests = this.props.pullRequests
+        if (pullRequests) {
+          if (pullRequests.length > 0) {
+            return (
+              <PullRequestList
+                pullRequests={pullRequests}
+                onPullRequestClicked={this.onPullRequestClicked}
+                onDismiss={this.onDismiss}
+              />
+            )
+          } else {
+            const repo = this.props.repository
+            const name = repo.gitHubRepository
+              ? repo.gitHubRepository.fullName
+              : repo.name
+            const isOnDefaultBranch =
+              this.props.defaultBranch &&
+              this.props.currentBranch &&
+              this.props.defaultBranch.name === this.props.currentBranch.name
+            return (
+              <NoPullRequests
+                repositoryName={name}
+                isOnDefaultBranch={!!isOnDefaultBranch}
```

Comment:
Why the double `!!` here?

Reply:
`isOnDefaultBranch` is `boolean | null`, but the `isOnDefaultBranch` prop is `boolean`. So the double negation gets rid of the `null`.